metals/src/main/scala/scala/meta/internal/metals/ScalaVersions.scala

```diff
@@ -60,15 +60,15 @@ object ScalaVersions {
     latestBinaryVersionFor(scalaVersion)
       .map(latest =>
         latest != scalaVersion && SemVer
-          .isCompatibleVersion(latest, scalaVersion)
+          .isLaterVersion(latest, scalaVersion)
```

Comment:
This was the main issue. `3.0.0-M1` was equal to `3.0.0-M2`, because we were comparing only major/minor/patch. 
First of all it should not say it's a future version if it's equal, second of all it would be good to support also milestones/release candidates now.